aspnetcore/publishing/linuxproduction.md

```diff
@@ -18,75 +16,99 @@ uid: publishing/linuxproduction
- - In a browser, navigate to `http://<serveraddress>:<port>` to verify you app works on Linux. 
+Copy the ASP.NET Core app to the server using whatever tool (SCP, FTP, etc.) integrates into your workflow. Test the app, for example:
+ - From the command line, run `dotnet yourapp.dll`
+ - In a browser, navigate to `http://<serveraddress>:<port>` to verify the app works on Linux. 
  
-Note: You can use [Yeoman](../client-side/yeoman.md) to create a new ASP.NET Core application for a new project.
+**Note:** Use [Yeoman](xref:client-side/yeoman) to create a new ASP.NET Core app for a new project.
 
 ## Configure a reverse proxy server
 
-A reverse proxy is a common setup for serving dynamic web applications. The reverse proxy terminates the HTTP request and forwards it to the ASP.NET application.
+A reverse proxy is a common setup for serving dynamic web applications. The reverse proxy terminates the HTTP request and forwards it to the ASP.NET Core application.
+
+### Why use a reverse proxy server?
 
-### Why use a reverse-proxy server?
+Kestrel is great for serving dynamic content from ASP.NET Core; however, the web serving parts aren’t as feature rich as servers like IIS, Apache, or Nginx. A reverse proxy server can offload work like serving static content, caching requests, compressing requests, and SSL termination from the HTTP server. The reverse proxy server may reside on a dedicated machine or may be deployed alongside an HTTP server.
 
-Kestrel is great for serving dynamic content from ASP.NET, however the web serving parts aren’t as feature rich as full-featured servers like IIS, Apache or Nginx. A reverse proxy-server can allow you to offload work like serving static content, caching requests, compressing requests, and SSL termination from the HTTP server. The reverse proxy server may reside on a dedicated machine or may be deployed alongside an HTTP server.
+For the purposes of this guide, a single instance of Nginx is used. It runs on the same server, alongside the HTTP server. Based on your requirements, you may choose a different setup.
 
-For the purposes of this guide, we are going to use a single instance of Nginx that runs on the same server alongside your HTTP server. However, based on your requirements you may choose a different setup.
+Because requests are forwarded by reverse proxy, use the `ForwardedHeaders` middleware from the `Microsoft.AspNetCore.HttpOverrides` package. This middleware updates `Request.Scheme`, using the `X-Forwarded-Proto` header, so that redirect URIs and other security policies work correctly.
 
-When setting up a reverse-proxy server other than IIS, you must call `app.UseIdentity` (in `Configure`) before any other external providers.
+When setting up the reverse proxy server, the authentication middleware needs `UseForwardedHeaders` to run first. This ordering ensures that the authentication middleware can consume the affected values and generate correct redirect URIs.
```

Comment:
Nit: There is a lot of switching back-and-forth between "a reverse proxy" and "the reverse proxy". I would personally use "a reverse proxy" here, but I ain't no English expert, so I could easily be wrong on this.